### PR TITLE
refactor(mutagen): always use `getMutagenEnv` helper to define Mutagen env

### PR DESCRIPTION
**Pull request by vvagaytsev**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
A tiny predecessor for #5551.

### CODE REVIEW

File: core/src/mutagen.ts [hunk 1]
```diff
@@ -808,10 +805,30 @@ export function getMutagenDataDir(path: string, log: Log) {
...
+  [k in MutagenRequiredEnvVar]: string
+}
+
+const mutagenOptionalEnvVars = ["MUTAGEN_LOG_LEVEL"] as const
+type MutagenOptionalEnvVar = (typeof mutagenOptionalEnvVars)[number]
+type MutagenOptionalEnv = {
+  [k in MutagenOptionalEnvVar]?: string
+}
+
+type MutagenEnv = MutagenRequiredEnv & MutagenOptionalEnv
```
Comment (shumailxyz) [inline core/src/mutagen.ts:820]: I'm wondering if we can merge these 2 types into 1 as we are currently only using `MUTAGEN_DATA_DIRECTORY ` and `MUTAGEN_LOG_LEVEL`. Do we need this flexibility and separate types for required/optional or its too verbose atm?

**Review by shumailxyz — APPROVED** [review]
Minor comment, but LGTM otherwise. Thanks :)

File: core/src/mutagen.ts (hunk 1 above)
Comment (vvagaytsev) [reply core/src/mutagen.ts]: Good point, thank you! Simplified this in d82f9f97b5d309001ad166862b67eb6b2ab0b9ff.

**Review by shumailxyz — APPROVED** [review]
Awesome ✨

**Pull request merged**